### PR TITLE
X3D serializer: optional smoothing, Y up, X3D version

### PR DESCRIPTION
**Pull request by andreasplesch**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

addresses partially #1235
also sets X3D version to 3.3 since some printing services check for the version

### CODE REVIEW

File: packages/io/x3d-serializer/src/index.js [hunk 1]
```diff
@@ -190,6 +190,7 @@ const convertGeom3 = (object, options) => {
   const shape = ['Shape', {}, convertMesh(object, options)]
   if (object.color) {
     shape.push(convertAppearance(object, options))
+    shape[3][1][1].emissiveColor="0 0 0"
```
Comment (andreasplesch) [inline packages/io/x3d-serializer/src/index.js]: This is just an initial hack. I would have a better solution in another PR.

**Comment by andreasplesch** [comment]
https://gist.githack.com/andreasplesch/dc3595c5cf5198bedd7dbdb119a7a059/raw/4c9aed0dd47191873fc4e87870125eecc75c1ab1/x3dom_jscad.html is how the X3D looks after these changes.

File: packages/io/x3d-serializer/src/index.js [hunk 2]
```diff
@@ -128,7 +130,7 @@ const convertObjects = (objects, options) => {
       shapes.push(convertPath2(object, options))
     }
   })
-  scene = scene.concat(shapes)
+  scene[1] = scene[1].concat(shapes)
```
Comment (andreasplesch) [inline packages/io/x3d-serializer/src/index.js:133]: perhaps simplify to:
```js
const scene = [ 'Scene', [ 'Transform', { rotation: '1 0 0 -1.5708' }, shapes ] ]
```

File: packages/io/x3d-serializer/src/index.js [hunk 3]
```diff
@@ -48,6 +48,7 @@ const mimeType = 'model/x3d+xml'
  * Serialize the give objects to X3D elements (XML).
  * @param {Object} options - options for serialization, REQUIRED
  * @param {Array} [options.color=[0,0,1,1]] - default color for objects
+ * @param {Boolean} [options.smooth=false] - use averaged vertex normals
```
Comment (z3dev) [inline packages/io/x3d-serializer/src/index.js:51]: Cool!

File: packages/io/x3d-serializer/src/index.js (hunk 2 above)
Comment (z3dev) [reply packages/io/x3d-serializer/src/index.js:133]: I kind of like the second option as that shows the hierarchy between scene and transform.

File: packages/io/x3d-serializer/src/index.js [hunk 4]
```diff
@@ -202,7 +205,7 @@ const convertMesh = (object, options) => {
 
   const faceset = [
     'IndexedTriangleSet',
-    { ccw: 'true', colorPerVertex: 'false', solid: 'false', index: indexList },
+    { ccw: 'true', colorPerVertex: 'false', normalPerVertex: options.smooth === true, solid: 'false', index: indexList },
```
Comment (z3dev) [inline packages/io/x3d-serializer/src/index.js:208]: Maybe options.smooth is enough. Typically, the values of options are not checked but... correct values are expected.

File: packages/io/x3d-serializer/src/index.js [hunk 5]
```diff
@@ -188,6 +190,7 @@ const convertGeom3 = (object, options) => {
   const shape = ['Shape', {}, convertMesh(object, options)]
   if (object.color) {
     shape.push(convertAppearance(object, options))
+    shape[3][1][1].emissiveColor = '0 0 0'
```
Comment (z3dev) [inline packages/io/x3d-serializer/src/index.js:193]: Anyway to make this more intuitive? Maybe convertMesh can add emissiveColor?

**Review by z3dev — COMMENTED** [review]
@andreasplesch Nice.

I just have a few comments but these changes are very solid. Thanks again.

File: packages/io/x3d-serializer/src/index.js (hunk 2 above)
Comment (andreasplesch) [reply packages/io/x3d-serializer/src/index.js:133]: will change.

File: packages/io/x3d-serializer/src/index.js (hunk 4 above)
Comment (andreasplesch) [reply packages/io/x3d-serializer/src/index.js:208]: will change to options.smooth .

File: packages/io/x3d-serializer/src/index.js (hunk 5 above)
Comment (andreasplesch) [reply packages/io/x3d-serializer/src/index.js:193]: The idea would be to add emissiveColor for Lines, and only diffuseColor for Meshes. I would like to add a parameter to convertAppearance() . Let me update if this sounds acceptable.

**Review by z3dev — CHANGES_REQUESTED** [review]
@andreasplesch just a few very small requests. otherwise, this looks ready to go.

File: packages/io/x3d-serializer/src/index.js [hunk 6]
```diff
@@ -61,6 +62,7 @@ const mimeType = 'model/x3d+xml'
 const serialize = (options, ...objects) => {
   const defaults = {
     color: [0, 0, 1, 1.0], // default colorRGBA specification
+    smooth: false,
     decimals: 1000,
```
[the remark this answers is not in the document]
Comment (andreasplesch) [reply packages/io/x3d-serializer/src/index.js:66]: Will do.

File: packages/io/x3d-serializer/src/index.js [hunk 7]
```diff
@@ -262,10 +267,12 @@ const polygons2coordinates = (polygons, options) => {
...
       // add the vertex to the list of points (and index) if not found
       if (!vertexTagToCoordIndexMap.has(id)) {
-        const x = Math.round(vertex[0] * options.decimals) / options.decimals
-        const y = Math.round(vertex[1] * options.decimals) / options.decimals
-        const z = Math.round(vertex[2] * options.decimals) / options.decimals
+        const x = roundToDecimals(vertex[0], options)
+        const y = roundToDecimals(vertex[1], options)
+        const z = roundToDecimals(vertex[2], options)
         pointList.push(`${x} ${y} ${z}`)
+        //const roundedVertex = vertex.map((v) => roundToDecimals(v, options))
```
[the remark this answers is not in the document]
Comment (andreasplesch) [reply packages/io/x3d-serializer/src/index.js:274]: not used.

**Review by z3dev — APPROVED** [review]
@andreasplesch nice. thanks for these fixes.

**Comment by andreasplesch** [comment]
Super.

**Pull request merged**